### PR TITLE
Remove Address Book Global Variables

### PR DESCRIPTION
**Pull request by RomeuG**
* **What does this PR do?**

This PR eliminates Alias configuration global variables.

* **What are the relevant issue numbers?**
#2621

### CODE REVIEW

**Comment by RomeuG** [comment]
@flatcap 

Hey!

So I did my best on removing `NeoMutt->sub` dependency from `Alias` directory.
Had to change some function signature to receive `struct ConfigSubset`.

You can also see that some functions in `Alias` are used outside of it. In those cases, and if they have a `struct ConfigSubset` as a parameter`, I pass `NeoMutt->sub`.

A new C file was created (`alias/sort.c`), which has the function signatures in `alias/lib.h` since they are used in other files.

Thanks!

File: alias/sort.c [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    return;
+
+  sort_alias = cs_subset_sort(sub, "sort_alias");
+  ARRAY_SORT(ava, alias_get_sort_function(sort_alias));
+
+  struct AliasView *avp = NULL;
+  int i = 0;
+  ARRAY_FOREACH(avp, ava)
+  {
+    avp->num = i++;
```
Comment (flatcap) [inline alias/sort.c:134]: We can drop the `i` here.
If you look inside `ARRAY_FOREACH()`, then inside the macro inside, you'll find an array loop variable that we can use.

**Review by flatcap — CHANGES_REQUESTED** [review]
I've only looked at the sort changes, so far.
We can go a little further.

File: alias/sort.c [hunk 2]
```diff
@@ -0,0 +1,136 @@
+#include "alias.h"
```
[the remark this answers is not in the document]
Comment (RomeuG) [reply alias/sort.c:30]: Me and the so called distraction go hand in hand!
Fixed!

File: alias/sort.c [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  int r = (av_a->orig_seq - av_b->orig_seq);
+
+  return RSORT(r);
+}
+
+/**
+ * alias_get_sort_function - Sorting function decision logic
+ * @param sort Sort method, e.g. #SORT_ALIAS
+ */
+sort_t alias_get_sort_function(short sort)
```
[the remark this answers is not in the document]
Comment (RomeuG) [reply alias/sort.c]: I obviously did eliminate the call, but I did keep the function. I think it is more organized...
If you feel that it should be merged, just tell me and I'll remove the function.

File: alias/sort.c (hunk 1 above)
Comment (RomeuG) [reply alias/sort.c]: Used it before, dunno why I didn't use there.

**Comment by flatcap** [comment]
By the magic of `git`, I've separated out and merged your `alias/sort.c` changes (bb7d3bbb7), and rebased this branch.

**Review by flatcap — APPROVED** [review]

**Pull request merged**